assets/web-ui/js/widget.js

```diff
@@ -1703,6 +1720,10 @@ export class FlatProfile extends Widget {
         for (let i = 0; i < limit; i++) {
             const stats = functionsStats[i];
 
+            if (this.searchQuery && (stats.functionName.toLowerCase()).indexOf(this.searchQuery.toLowerCase()) < 0) {
```

Comment:
This is IMO not the expected behavior. I mean that this feature should only highlight the functions that match the search pattern rather than filter them.
It would also be consistent with what it is done by other widgets.

Reply:
@NoiseByNorthwest 
Ok maybe, it's because misplaced  search input and it should be part of the flat profile.
Technically we have category view that is way powerful that search/filter feature I wanted to introduce.
Sometimes on big timelines it's hard to find a function only using highlight, so I think it would be nice to have possibility to filter ones in the flat profile with a simple color change on other widgets.
Something like blackfire has in the CallGraph.
![Selection_681](https://user-images.githubusercontent.com/3192770/116969933-69442b80-acb7-11eb-904a-a389935bbcb8.png)

Reply:
OK I got it, thanks for the explanation.
Regarding the implementation here, it may be more readable if you apply the filtering before the first use of `functionsStats`, and also the table limit will properly be handled.

```js
const functionsStats = this
    .timeRangeStats
    .getFunctionsStats()
    .getValues()
    .filter(stats => !(
        this.searchQuery && (stats.functionName.toLowerCase()).indexOf(this.searchQuery.toLowerCase()) < 0
    ))
;
```